common/txmgr/types/storage_service.go

```diff
@@ -0,0 +1,95 @@
+	// FindTransactionsConfirmedInBlockRange(highBlockNumber, lowBlockNumber int64, chainID CHAINID, opts ...any) (etxs []*TX, err error)
+	// GetEthTxInProgress(fromAddress ADDR, opts ...any) (etx *TX, err error)
+	// GetInProgressEthTxAttempts(ctx context.Context, address ADDR, chainID CHAINID, opts ...any) (attempts []TXATTEMPT, err error)
+	// HasInProgressTransaction(account ADDR, chainID CHAINID, opts ...any) (exists bool, err error)
+	// // InsertEthReceipt only used in tests. Use SaveFetchedReceipts instead
+	InsertEthReceipt(receipt *Receipt[R, HASH], opts ...any) error
+	// InsertEthTx(etx *TX, opts ...any) error
+	// InsertEthTxAttempt(attempt *TXATTEMPT, opts ...any) error
+	// LoadEthTxAttempts(etx *TX, opts ...any) error
+	// LoadEthTxesAttempts(etxs []*TX, opts ...any) error
+	// MarkAllConfirmedMissingReceipt(chainID CHAINID, opts ...any) (err error)
+	// MarkOldTxesMissingReceiptAsErrored(blockNum int64, finalityDepth uint32, chainID CHAINID, opts ...any) error
+	// PreloadEthTxes(attempts []TXATTEMPT, opts ...any) error
+	PruneUnstartedEthTxQueue(queueSize uint32, subject uuid.UUID, opts ...any) (n int64, err error)
+	// SaveConfirmedMissingReceiptAttempt(ctx context.Context, timeout time.Duration, attempt *TXATTEMPT, broadcastAt time.Time, opts ...any) error
+	SaveFetchedReceipts(receipts []R, chainID CHAINID, opts ...any) (err error)
+	// SaveInProgressAttempt(attempt *TXATTEMPT, opts ...any) error
+	// SaveInsufficientEthAttempt(timeout time.Duration, attempt *TXATTEMPT, broadcastAt time.Time, opts ...any) error
+	// SaveReplacementInProgressAttempt(oldAttempt TXATTEMPT, replacementAttempt *TXATTEMPT, opts ...any) error
+	// SaveSentAttempt(timeout time.Duration, attempt *TXATTEMPT, broadcastAt time.Time, opts ...any) error
+	// SetBroadcastBeforeBlockNum(blockNum int64, chainID CHAINID, opts ...any) error
+	// UpdateBroadcastAts(now time.Time, etxIDs []TXID, opts ...any) error
+	// UpdateEthKeyNextNonce(newNextNonce, currentNextNonce TXMETA, address ADDR, chainID CHAINID, opts ...any) error
+	// UpdateEthTxAttemptInProgressToBroadcast(etx *TX, attempt TXATTEMPT, NewAttemptState TxAttemptState, incrNextNonceCallback CallbackFunc, opts ...any) error
+	// UpdateEthTxsUnconfirmed(ids []int64, opts ...any) error
+	// UpdateEthTxUnstartedToInProgress(etx *TX, attempt *TXATTEMPT, opts ...any) error
+	// UpdateEthTxFatalError(etx *TX, opts ...any) error
+	// UpdateEthTxForRebroadcast(etx TX, etxAttempt TXATTEMPT, opts ...any) error
+	// Close()
+}
+
+// TxStrategy controls how txes are queued and sent
+//
+//go:generate mockery --quiet --name TxStrategy --output ./mocks/ --case=underscore --structname TxStrategy --filename tx_strategy.go
+type TxStrategy interface {
+	// Subject will be saved to eth_txes.subject if not null
+	Subject() uuid.NullUUID
+	// PruneQueue is called after eth_tx insertion
+	PruneQueue(pruneService any, opt any) (n int64, err error)
```

Comment:
nit: Can you describe the params pruneService and opt in comments, since with their type removed, they just don't make any sense.

Reply:
Also, since our plan is to eventually not let TxMgr interfaces expose PostGres/DB specific QOpts params, its ok if you keep opt param as the actual type right now, and not convert to generic.

Reply:
Gotcha, I agree. Jordan mentioned something similar. Will fix